src/main/java/com/ksabbak/javaserver/storage/Persistable.java

```diff
@@ -0,0 +1,14 @@
+package com.ksabbak.javaserver.storage;
+
+public interface Persistable {
```

Comment:
I am curious. How did you arrive at the decision to throw exceptions for all these functions?

Reply:
So, I figured when someone calls these methods, they expect something to happen and if that thing doesn't happen there should be some indication that that's the case. Returning `null` seems like an especially bad option, I figured an Exception did what I needed it to without doing things that might get a little weird. I think there's probably a better option, considering I think querying a thing for something that isn't there doesn't seem "exceptional", but I couldn't come up with a better alternative.